forcing redraw of snapshot results when snapshot was pre-selected in …

etanshaul opened the pull request:
…pending state

Fixes #416 

The issue was that when the user selected a snapshot (by clicking on the row in the IJ debug window) prior to the snapshot being triggered, the results were not being redrawn when the data came in. Previously the stack frames and variables were only redrawn on a mouse click; and the mouse click is ignored if it was a click on an already selected row.

The solution is to force redraw the results when they come in if the row was _already_ selected before they came in.

etanshaul commented:
@patflynn @loosebazooka

patflynn commented on google-cloud-tools-plugin/src/com/google/gct/idea/debugger/CloudDebugProcess.java:
```diff
@@ -261,7 +261,7 @@ public void run() {
    * Finds the snapshot associated with the given id and sets it as the active snapshot in the current debug session.
    */
-  public void navigateToSnapshot(@NotNull final String id) {
+  public void navigateToSnapshot(@NotNull final String id, final boolean isSelectedBeforeTrigger) {
```
could you add javadoc for isSelectBeforeTrigger?  Typically boolean flags to a method to change its behavior is discouraged for public API.

elharo commented on google-cloud-tools-plugin/src/com/google/gct/idea/debugger/CloudDebugProcess.java:
```diff
@@ -286,7 +286,7 @@ public void run() {
                   }
 
-                  if (myCurrentSnapshot == null || !myCurrentSnapshot.getId().equals(result.getId())) {
+                  if (isSelectedBeforeTrigger || myCurrentSnapshot == null || !myCurrentSnapshot.getId().equals(result.getId())) {
```
what if we jsut did this unconditionally? i.e. removed the if () completely?

etanshaul replied:
@elharo we could do that. That would mean though that if you already have a snapshot selected, and you click on it again it would do a redundant reloading of the result.

etanshaul commented:
@patflynn, @elharo - I refactored the code a bit based on both of your comments:

- I moved the extra "override" boolean parameter out of the public API and into the caller.
- I also moved the conditional logic that prevents redundant reloads of snapshot data (when clicking on an already selected row) out to the caller

patflynn commented:
LGTM

patflynn commented:
merging to get this into our next release cut.
